### DISCUSSION TITLE
Import (gpt made) json recipes / json specification for recipes

### DISCUSSION THREAD

**Discussion by Ilosariph** [TandoorRecipes/recipes#3308] [Q&A] [answered]
I'm currently trying out tandoor, and I have a lot of recipes that are on paper.
I saw that tandoor has an export and import function, so I exported a sample recipe, and created a sample json. Gave that + a photo of the recipe to ChatGPT and got a (to me) good looking json. I then packaged it like it was before (recipe.json in a zip called 1.zip, and that in another zip file).

I couldn't find any specification in the docs, just that the default import exists. Importing a before exported file works fine, so the import function seems to work with a normal file. Comparing the file to the original I can't see any differences preventing this from working.

When I try importing, I get the following error:
```
-------------------- 
ERROR 
'NoneType' object has no attribute 'keywords'
--------------------
Imported 0 recipes.
```
Here is the recipe I've generated:
[recipe.json](https://github.com/user-attachments/files/17034001/recipe.json)

**Comment by Kiwifed0r** [comment, marked as answer]
The keywords are not in the right format. I think they need to look something like:

`"keywords":
    [
        {
            "name": "test",
            "description": "",
            "created_at": "2024-10-12T21:32:52.549700+02:00",
            "updated_at": "2024-10-15T13:43:17.444207+02:00"
        }
    ]`

And there are some issues with units and amounts. I attached a working version:
[recipe.json](https://github.com/user-attachments/files/17390547/recipe.json)

This also solves my issue described in #3366. Thank you for the inspiration!

**Reply by Ilosariph** [reply]
Good catch, thanks. Let me know if you figure out a consistently working prompt. I'll have another go sometime.

**Reply by mr-woodapple** [reply]
I tried importing the .json you attached, but it's not working for me - neither the tags, nor the steps are imported. Any obvious idea why? (Came here after trying to import a receipe I created via the ui after getting the json format from the network tab, which also wasn't working).

**Reply by ZD-FLoM** [reply]
Also not working for me - the working version posted by @Kiwifed0r , at least in the latest Tandoor beta version

**Reply by jstrvr** [reply]
In perplexity I've made a space (something similar to a GPT) to convert anything to the tandoor format. I've attached the following recipeformat. [recipeformat.txt](https://github.com/user-attachments/files/21170552/recipeformat.txt)
I've tried only taking the script part and that also seems to work.

I've given the space the following prompt.
[tandoor converter.txt](https://github.com/user-attachments/files/21170597/tandoor.converter.txt)

I now only have to put in an image or link and type "convert this file using recipeformat.txt". This seems to work pretty well. When importing I split the steps and automatically sort ingredients. Any improvements to the recipe format or tandoor prompt are welcome.

**Comment by johnmacdowall** [comment]
Yeah, this is hard!  I've been reverse engineering mine for, what's today?, Friday?, four days!  IF anybody's got a fully fleshed out JSON that Tandoor ingests into all possible fields, that would be muchly appreciated!!!  I'll make sure to implement your tags methodology today.  THX

**Comment by nprime496** [comment]
Same issue as @johnmacdowall the import is not consistent, really hard to know that is expected, at this point, a validator may be helpful.
